common/src/main/java/com/microsoft/identity/common/crypto/AndroidBrokerStorageEncryptionManager.java

```diff
@@ -0,0 +1,173 @@
+            return mKeyStoreKeyLoader;
+        }
+
+        if (AZURE_AUTHENTICATOR_APP_PACKAGE_NAME.equalsIgnoreCase(getPackageName())) {
+            return mLegacyAuthAppKeyLoader;
+        }
+
+        if (COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(getPackageName()) ||
+                BROKER_HOST_APP_PACKAGE_NAME.equalsIgnoreCase(getPackageName())) {
+            return mLegacyCPKeyLoader;
+        }
+
+        throw new IllegalStateException("Matching encryption key not found");
+    }
+
+    @Override
+    public @NonNull List<AbstractSecretKeyLoader> getKeyLoaderForDecryption(@NonNull byte[] cipherText) {
+        final String methodName = ":getKeyLoaderForDecryption";
+        final String packageName = getPackageName();
+
+        if (isEncryptedByThisKeyIdentifier(cipherText, PredefinedKeyLoader.KEY_IDENTIFIER)) {
+            if (COMPANY_PORTAL_APP_PACKAGE_NAME.equalsIgnoreCase(packageName) ||
+                    BROKER_HOST_APP_PACKAGE_NAME.equalsIgnoreCase(packageName)) {
+                return new ArrayList<AbstractSecretKeyLoader>() {{
+                    add(mLegacyCPKeyLoader);
+                    add(mLegacyAuthAppKeyLoader);
+                }};
+            } else if (AZURE_AUTHENTICATOR_APP_PACKAGE_NAME.equalsIgnoreCase(packageName)) {
+                return new ArrayList<AbstractSecretKeyLoader>() {{
+                    add(mLegacyAuthAppKeyLoader);
+                    add(mLegacyCPKeyLoader);
+                }};
+            } else {
+                throw new IllegalStateException("Unexpected Broker package name. Cannot load key.");
+            }
+        }
+
+        if (isEncryptedByThisKeyIdentifier(cipherText, AndroidWrappedKeyLoader.KEY_IDENTIFIER)) {
+            return new ArrayList<AbstractSecretKeyLoader>() {{
+                add(mKeyStoreKeyLoader);
+            }};
+        }
+
+        Logger.warn(TAG + methodName, "Cannot find a matching key to decrypt the given blob");
+        return new ArrayList<>();
+    }
+
+    @Override
+    protected void handleDecryptionFailure(@NonNull final String keyAlias,
+                                           @NonNull final Exception exception) {
+        final String methodName = ":handleDecryptionFailure";
+        final SharedPreferences sharedPreferences = PreferenceManager.
+                getDefaultSharedPreferences(mContext);
+        final String previousActiveBroker = sharedPreferences.getString(
+                CURRENT_ACTIVE_BROKER_SHARED_PREF_KEY,
+                ""
+        );
+        final String activeBroker = mContext.getPackageName();
+
+        // We don't want to emit the same value multiple time.
+        if (!previousActiveBroker.equalsIgnoreCase(activeBroker)) {
+            final String message = "Decryption failed with key: " + keyAlias
+                    + " Active broker: " + activeBroker
+                    + " Exception: " + exception.toString();
+
+            Logger.info(TAG + methodName, message);
```

Comment:
FYI keys can contain PII/OII like RP names, tenants, and OIDs

Reply:
This is keyAlias (the name we hardcoded to the key - see ISecretKeyLoader.getKeyAlias()) , not the key itself.